src/test/java/com/ibm/cloud/sdk/core/test/security/ConfigBasedAuthenticatorFactoryTest.java

```diff
@@ -201,6 +278,12 @@ public void testEnvCredentialsService1() {
     Authenticator auth = ConfigBasedAuthenticatorFactory.getAuthenticator("service-1");
     assertNotNull(auth);
     assertEquals(Authenticator.AUTHTYPE_IAM, auth.authenticationType());
+    IamAuthenticator iamAuth = (IamAuthenticator) auth;
```

Comment:
Noticed that we weren't doing much checking for the IAM and CP4D authenticators here so added some extra validation.